packages/headless/src/controllers/insight/attach-to-case/headless-attach-to-case.ts

```diff
@@ -0,0 +1,129 @@
+import {ArrayValue, RecordValue, Schema} from '@coveo/bueno';
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {configuration, attachedResults} from '../../../app/reducers';
+import {
+  attachResult,
+  detachResult,
+} from '../../../features/attached-results/attached-results-actions';
+import {AttachedResult} from '../../../features/attached-results/attached-results-state';
+import {
+  AttachedResultsSection,
+  ConfigurationSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  nonEmptyString,
+  requiredNonEmptyString,
+  validateInitialState,
+} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+export interface AttachToCaseState {
+  attachedResults: AttachedResult[];
+}
+
+export interface AttachToCaseInitialState extends AttachToCaseState {}
+
+export interface AttachToCaseProps {
+  initialState?: AttachToCaseInitialState;
+}
+
+export interface SearchResult {
+  title: string;
+  raw: {
+    permanentid?: string;
+    urihash?: string;
+  };
+}
+
+export interface AttachToCase extends Controller {
+  isAttached(result: SearchResult): boolean;
+  attach(result: AttachedResult): void;
+  detach(result: AttachedResult): void;
+}
+
+const initialStateSchema = new Schema({
+  attachedResults: new ArrayValue({
+    required: false,
+    each: new RecordValue({
+      options: {
+        required: true,
+      },
+      values: {
+        knowledgeArticleId: nonEmptyString,
+        articleLanguage: nonEmptyString,
+        articleVersionNumber: nonEmptyString,
+        articlePublishStatus: nonEmptyString,
+        uriHash: nonEmptyString,
+        permanentId: nonEmptyString,
+        resultUrl: requiredNonEmptyString,
+        source: requiredNonEmptyString,
+        title: requiredNonEmptyString,
+      },
+    }),
+  }),
+});
+
+export function buildAttachToCase(
+  engine: InsightEngine,
+  props: AttachToCaseProps = {}
+): AttachToCase {
+  if (!loadAttachedResultsReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  validateInitialState(
```

Comment:
Are you setting the initial state anywhere? Looks like your just validating it.
A common pattern in headless controllers with an initial state is to "register" them whent he controller is built.
You can see [headless-recent-result-list](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/controllers/recent-results-list/headless-recent-results-list.ts) as an example. 
I see you already have an action called setAttachedResults. My suggestion would be to rename it to `registerAttachedResults` to be a bit more consistend and dispatch the action right after validating the initial state. On the UI side there won't be any need to load and dispatch the action to register the exisiting results.